dbldatagen/data_generator.py

```diff
@@ -131,9 +132,53 @@ def __init__(self, sparkSession=None, name=None, randomSeedMethod=None,
         self.withColumn(ColumnGenerationSpec.SEED_COLUMN, LongType(), nullable=False, implicit=True, omit=True)
         self._batchSize = batchSize
 
+        # set up spark session
+        self._setupSparkSession(sparkSession)
+
         # set up use of pandas udfs
         self._setupPandas(batchSize)
 
+    @classmethod
+    def _checkSparkVersion(cls, sparkVersion, minSparkVersion):
+        """
+        check spark version
+        :param sparkVersion: spark version string
+        :param minSparkVersion: min spark version as tuple
+        :return: True if version passes minVersion
+
+        Layout of version string must be compatible "xx.xx.xx.patch"
+        """
+        sparkVersionInfo = _get_spark_version(sparkVersion)
+
+        if sparkVersionInfo < minSparkVersion:
+            logging.warn(f"*** Minimum version of Python supported is {minSparkVersion} - found version %s ",
+                         sparkVersionInfo )
+            return False
+
+        return True
+
+    def _setupSparkSession(self, sparkSession):
+        """
+        Set up spark session
+        :param sparkSession: spark session to use
+        :return: nothing
+        """
+        if sparkSession is None:
+            sparkSession = SparkSingleton.getInstance()
+
+        self.sparkSession = sparkSession
+        if sparkSession is None:
+            raise DataGenError("""Spark session not initialized
+
```

Comment:
> Added line #L171 was not covered by tests

and this statement is unreachable :) lgtm.com is soo good at this.
<img width="628" alt="image" src="https://user-images.githubusercontent.com/259697/203790682-7e6ab788-341e-435e-ae38-f75d4cf1c84c.png">

Reply:
> and this statement is unreachable :) lgtm.com is soo good at this.

except the fix for this was checked in 20 hours earlier

Reply:
for some reason lgtm is taking 1 day to update results - perhaps due to transition to new code scanning ?